### DISCUSSION TITLE
writing plugins

### DISCUSSION THREAD

**Discussion by axiomgraph** [olive-editor/olive#2120] [General]
things I need to know for writing glsl plugins?

**Comment by eliramalho** [comment]
anxious for seeing your plugins!

**Reply by axiomgraph** [reply]
I don't understand why do you feel anxious for seeing my plugins

**Comment by AngeloFrancabandiera** [comment]
> things I need to know for writing glsl plugins?

So far Olive does not allow to write GLSL plugins, but there is a [pull request](https://github.com/olive-editor/olive/pull/1839) that will allow to.
Of course you need basic knowledge of GLSL language, but you may get your feet wet with [shader toy](http://www.shadertoy.com).
But the best will be to see examples written by others.

**Reply by axiomgraph** [reply]
I saw your pr before. a youtuber talking about it. I thought the admins will like it and they will merge. which did not happen. I already wrote glsl plugins for another program https://github.com/axiomgraph/enve-plugins . when I saw olive I thought I can rewrite those codes to work with olive. that's why I asked what are the requirements for writing glsl plugins. some of my glsl codes are from shadertoy.

**Reply by AngeloFrancabandiera** [reply]
Sorry, I didn't get your question. For writing plugins in Olive, there is a dedicated [wiki page](https://github.com/AngeloFrancabandiera/olive/wiki/Shader-filter) with all the explanation and samples.

**Reply by axiomgraph** [reply]
thanks let me check and I will come back later.

**Reply by eliramalho** [reply]
When my students, who are newbies, need to understand how things work, plugins prove to be extremely helpful

**Reply by eliramalho** [reply]
I lack the knowledge of executing it in an advanced manner.

**Reply by eliramalho** [reply]
I love teaching how to develop videos with your Program Olive

**Reply by eliramalho** [reply]
I have a passion for teaching the art of video development using your Program Olive version 1. It excels in its simplicity and lightweight nature, while still maintaining a professional approach.

**Comment by axiomgraph** [comment]
> I have a passion for teaching the art of video development using your Program Olive version 1. It excels in its simplicity and lightweight nature, while still maintaining a professional approach.
okey
